CERT-Polska/mquery
# Update dependencies

**Your checklist for this pull request**

This PR consists of updates to urgent, vulnerable packages mquery uses. In more details, I've done the following:  
- Replace `react-js-pagination` with `replace-js-pagination`. Yeah, this one is kinda stupid..  
Just like it says in the [npmjs page](https://www.npmjs.com/package/replace-js-pagination), it is derived from the package currently used and from my testing has the same interface. It requires minimal code change and [just works](https://media.tenor.com/rkI1a8s2Z6QAAAAC/todd-howard-it-just-works.gif) from my testing.
- Replace `react-multiselect-checkboxes` with `react-select`. The former seems abandoned and not updated for very long, and it also says on its npmjs page that it's based on `react-select` (which is actively maintained). Like the bullet above, the currently used package is based on the latter so changing required minimal code changes (as seen in diff).
- Updated `axios` to newest version.
- Moved `react-scripts` to `devDependencies`. This is actually recommended by CRA team, they give multi-paragraph explanation as to why it should be done. A bit sus, but reasonable nonetheless
- Ran `npm audit fix` and let it do it's thing.
- Removed unused `merge` package

This makes it so `npm audit --omit=dev` finds 0 vulnerabilities.

**Test plan**

My test plan was as follows:
- Set up dev env
- Make multiple files in `/samples`
- Index each with a different taint
- Make multiple YARA queries, check results make sense:
  - One without any taint selected that should not give any results
  - One with some taints that should not give any results
  - One without any taint that should bring some results
  - One with some taints that should bring some results
- Make enough YARA queries so that there are multiple pages in `recent jobs` page in mquery. Make sure pagination works

**Closing issues**

fixes https://github.com/CERT-Polska/mquery/issues/299
